src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
Should we wait for all requests to complete then before piping the results to the controller? That's the logical conclusion of using concat. What happens when one service responds in 5ms but the second service only responds in 50 seconds?

Further more, the filtering logic for "are there any changes to this endpoint that should result in a re-bind or re-render" would have to sit in the controller, and not the service. There's a lot of benefit to treating each endpoint as a unit in the controller vs a bulk.